ui/views.py

```diff
@@ -107,17 +107,19 @@ class VideoViewSet(viewsets.ModelViewSet):
     def destroy(self, request, *args, **kwargs):  # pylint: disable=unused-argument
         video = self.get_object()
         if request.GET.get("s3"):
-            video.s3_object.delete()
+            for file in video.videofile_set.all():
+                file.s3_object.delete()
         self.perform_destroy(video)
         return Response(status=status.HTTP_204_NO_CONTENT)
 
-    @detail_route(permission_classes=(IsAdminOrHasMoiraPermissions,))
-    def signed_url(self, request, pk=None):  # pylint: disable=unused-argument
+    @detail_route(permission_classes=(IsAdminOrHasMoiraPermissions,), url_path=r'signed_url/(?P<encoding>\w+)')
+    def signed_url(self, request, pk=None, encoding=None):  # pylint: disable=unused-argument
```

Comment:
I do not understand this: what is `pk` and why both `pk` and `encoding` are defaulted to `None`?